### DISCUSSION TITLE
Does Pipewire work on LXQt?

### DISCUSSION THREAD

**Discussion by YamiYukiSenpai** [lxqt/lxqt#2043] [Q&A] [answered]
Does LXQt currently work with Pipewire?

I have installed it recently on Void Linux, and I've been trying to switch, [but I'm having trouble](https://reddit.com/r/voidlinux/comments/oru6k2/help_setting_up_pipewire/).

**Comment by stefonarch** [comment]
I don'think LXQt does influence that, but I read:

> I put these 2 lines on .xinitrc, but they still didn't work. Am I supposed to be using one of them or both?

```
pipewire
pipewire-pulse
```
Not sure if  `~/.xinitrc` is a good place and probably you should use the format `exec=comand`, but anyway for user applications' autostart use LXQt settings > session > autostart is what is recommended also in the document from Void.

Try the comand(s) manually first, to make sure that this works.

**Reply by YamiYukiSenpai** [reply]
```
$ pulseaudio -k
$ pipewire-pulse 
[W][00092.873282][  module-rtkit.c:  200 translate_error()] RTKit error: org.freedesktop.DBus.Error.Failed
[W][00092.873297][  module-rtkit.c:  460 set_nice()] could not set nice-level to -11: I/O error                       
[W][00092.881595][        server.c:  568 start_unix_server()] server 0x55bf5b1b9ca0: socket '/run/user/1000/pulse/native' is in use                                              
[W][00092.881626][        server.c:  967 servers_create_and_start()] pulse-server 0x55bf5b1b9860: failed to start server on 'unix:/run/user/1000/pulse/native': Address in use   
[E][00092.881640][  pulse-server.c: 5045 pw_protocol_pulse_new()] pulse-server 0x55bf5b1b9860: no servers could be started: Address in use                                       
[E][00092.881656][          conf.c:  332 load_module()] config 0x7f91e3d19680: could not load mandatory module "libpipewire-module-protocol-pulse": Address in use               
[E][00092.881772][      pipewire.c:  117 main()] failed to create context: Address in use   
$ pipewire
[W][00198.893565][  module-rtkit.c:  200 translate_error()] RTKit error: org.freedesktop.DBus.Error.Failed            
[W][00198.893581][  module-rtkit.c:  460 set_nice()] could not set nice-level to -11: I/O error                       
[W][00198.904222][  module-rtkit.c:  200 translate_error()] RTKit error: org.freedesktop.DBus.Error.Failed            
[W][00198.904239][  module-rtkit.c:  633 impl_acquire_rt()] could not make thread realtime: I/O error                 
[W][00198.907337][  module-rtkit.c:  200 translate_error()] RTKit error: org.freedesktop.DBus.Error.Failed            
[W][00198.907351][  module-rtkit.c:  460 set_nice()] could not set nice-level to -11: I/O error                       
[W][00198.910535][  module-rtkit.c:  200 translate_error()] RTKit error: org.freedesktop.DBus.Error.Failed            
[W][00198.910548][  module-rtkit.c:  633 impl_acquire_rt()] could not make thread realtime: I/O error  
```

I just noticed that every time I kill it, it keeps coming back!

According to my trusty KSysGuard, the parent process of PulseAudio was `lxqt-session`, but I already disabled it on LXQt's autostart. Why is that?

**Reply by Chiitoo** [reply]
As others have mentioned, it should work just fine, but indeed, whatever keeps starting PulseAudio needs to stop.  I think PulseAudio is pretty quick to start by itself with just some application using audio, and if memory serves me right-like, there's a file where you can specifically tell it to not do that, though that might be an ages old tactic.

I'm not sure if parts of PulseAudio are still needed for PipeWire on Void Linux, so completely uninstalling it might not be an option.

(I never really use PulseAudio myself, but I've tested OBS Studio builds with it, and same goes with PipeWire now (I also don't use systemd, on Gentoo Linux).)

**Reply by YamiYukiSenpai** [reply]
I've solved it, and @Chiitoo gave me the lead to look into how to stop PulseAudio from autostarting

**Comment by tsujan** [comment]
> the parent process of PulseAudio was lxqt-session

Then your system is different from most. PulseAudio is controlled by `systemd`.

I'm not familiar with Pipewire but I think whatever solution you have for it under other DEs should also work under LXQt.

**Reply by YamiYukiSenpai** [reply]
Void doesn't use systemd

**Comment by YamiYukiSenpai** [comment, marked as answer]
Solved:

* I needed to edit `/etc/pulse/client.conf`, and set `autospawn = off`.
* I used LXQt's session settings, and added both `pipewire` & `pipewire-pulse`.

I believe the first one is the necessary one, and should've been mentioned in the documentation.  I found the answer in [Ask Ubuntu](https://askubuntu.com/a/10307).  I was told in [LXQt's discussion page in GitHub](https://github.com/lxqt/lxqt/discussions/2043#discussioncomment-1052038) to use LXQt's session settings, but I think adding `pipewire` & `pipewire-pulse` in `.xinitrc` would produce similar results. 

Slightly off-topic, Void's LXQt doesn't come with `pavucontrol-qt`, and had to install manually.
